app/api/tool_utilities.py

```diff
@@ -14,18 +14,21 @@ def load_config():
     with open(config_path, 'r') as f:
         return json.load(f)
 
-tools_config = load_config()
+#tools_config = load_config()
 
 def get_executor_by_name(module_path):
     try:
         module = __import__(module_path, fromlist=['executor'])
-        return getattr(module, 'executor')
+        executor = getattr(module, 'executor')
+        return executor
     except Exception as e:
         logger.error(f"Failed to import executor from {module_path}: {str(e)}")
         raise ImportError(f"Failed to import module from {module_path}: {str(e)}")
 
 def load_tool_metadata(tool_id):
     logger.debug(f"Loading tool metadata for tool_id: {tool_id}")
+    
+    tools_config = load_config()
```

Comment:
Keep tools_config in DRY principle (dont repeat yourself)

Reply:
Fixed it.